src/browser/api/window.js

```diff
@@ -1076,14 +1081,29 @@ Window.close = function(identity, force, callback = () => {}) {
     handleForceActions(identity, force, 'close-requested', payload, defaultAction);
 };
 
+function disabledFrameUnsubDecorator(identity) {
+    const windowKey = genWindowKey(identity);
+    return function() {
+        let refCount = disabledFrameRef.get(windowKey) || 0;
+        if (refCount > 1) {
+            disabledFrameRef.set(windowKey, --refCount);
+        } else {
+            Window.enableFrame(identity);
+        }
+    };
+}
 
-Window.disableFrame = function(identity) {
-    let browserWindow = getElectronBrowserWindow(identity);
+Window.disableFrame = function(requestorIdentity, windowIdentity) {
+    const browserWindow = getElectronBrowserWindow(windowIdentity);
+    const windowKey = genWindowKey(windowIdentity);
 
     if (!browserWindow) {
         return;
     }
 
+    let dframeRefCount = disabledFrameRef.get(windowKey) || 0;
+    disabledFrameRef.set(windowKey, ++dframeRefCount);
+    subscriptionManager.registerSubscription(disabledFrameUnsubDecorator(windowIdentity), requestorIdentity, `disable-frame-${windowKey}`);
```

Comment:
Does the `subscriptionManager` work cross runtime? Say if apps A  B and C are all in diff runtimes and A and B both disable frame on C...